lua/luasnip/extras/postfix.lua

```diff
@@ -2,68 +2,56 @@ local snip = require("luasnip.nodes.snippet").S
 local events = require("luasnip.util.events")
 local extend_decorator = require("luasnip.util.extend_decorator")
 local node_util = require("luasnip.nodes.util")
+local util = require("luasnip.util.util")
 
 local matches = {
 	default = [[[%w%.%_%-%"%']+$]],
 	line = "^.+$",
 }
 
-local function generate_opts(match_pattern, user_callback)
-	return {
-		callbacks = {
-			[-1] = {
-				[events.pre_expand] = function(snippet, event_args)
-					local pos = event_args.expand_pos
-					-- [1]: returns table, gets text end-exclusive.
-					local line_to_cursor = vim.api.nvim_buf_get_text(
-						0,
-						pos[1],
-						0,
-						pos[1],
-						pos[2],
-						{}
-					)[1]
-					local postfix_match = line_to_cursor:match(match_pattern)
-						or ""
-					-- clear postfix_match-text.
-					vim.api.nvim_buf_set_text(
-						0,
-						pos[1],
-						pos[2] - #postfix_match,
-						pos[1],
-						pos[2],
-						{ "" }
-					)
-					local user_env = {}
-					if user_callback then
-						user_env = user_callback(snippet, event_args) or {}
-					end
-					local postfix_env_override = {
-						env_override = {
-							POSTFIX_MATCH = postfix_match,
-						},
-					}
+local function wrap_resolve_expand_params(
+	match_pattern,
+	user_condition,
+	user_resolve
+)
+	return function(snippet, line_to_cursor, match, captures)
+		if line_to_cursor:sub(1, -1 - #match):match(match_pattern) == nil then
+			return nil
+		end
 
-					return vim.tbl_deep_extend(
-						"keep",
-						user_env,
-						postfix_env_override
-					)
-				end,
-			},
-		},
-	}
-end
+		if user_condition then
+			if not user_condition(line_to_cursor, match, captures) then
+				return nil
+			end
+		end
```

Comment:
I think we can throw this out too now :D

Reply:
remove this `user_condition` support?

Reply:
Yeah, I think that handles `snip.condition`, and that is entirely inside the `resolve_expand_params`-callback now (ie. no longer set at all)

Reply:
Ah, i got it. fix this.